grpc-server-utils/src/main/java/org/hypertrace/core/grpcutils/server/ExternalExceptionInterceptor.java

```diff
@@ -0,0 +1,88 @@
+package org.hypertrace.core.grpcutils.server;
+
+import static io.grpc.Metadata.ASCII_STRING_MARSHALLER;
+import static org.hypertrace.core.grpcutils.context.ContextualStatusExceptionBuilder.from;
+
+import io.grpc.ForwardingServerCall.SimpleForwardingServerCall;
+import io.grpc.Metadata;
+import io.grpc.ServerCall;
+import io.grpc.ServerCall.Listener;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.grpc.Status;
+import java.util.Optional;
+import java.util.UUID;
+import org.hypertrace.core.grpcutils.context.ContextualExceptionDetails;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+import org.hypertrace.core.grpcutils.context.RequestContextConstants;
+
+/**
+ * This interceptor can be used at the edge to scrub any sensitive information such as an exception
+ * cause or metadata off the context before propagating it
+ */
+public class ExternalExceptionInterceptor implements ServerInterceptor {
```

Comment:
Should we define this as an abstract class so that one can define custom message and custom trailer building logic 🤔